util/tls/tls_socket.cc

```diff
@@ -428,15 +430,16 @@ error_code TlsSocket::HandleUpstreamWrite() {
       return write_result.error();
     }
     CHECK_GT(*write_result, 0u);
+
     upstream_write_ += *write_result;
     engine_->ConsumeOutputBuf(*write_result);
-    buffer.remove_prefix(*write_result);
-  }
 
-  if (engine_->OutputPending() > 0) {
-    LOG(INFO) << "ssl buffer is not empty with " << engine_->OutputPending()
-              << " bytes. short write detected";
-  }
+    // We could preempt while calling WriteSome, and the engine could get more data to write.
+    // Therefore we sync the buffer.
+    buffer = engine_->PeekOutputBuf();
+  } while (!buffer.empty());
+
+  DCHECK_EQ(engine_->OutputPending(), 0u);
```

Comment:
Should we also log this if the condition is not true just in case since `DCHECK` only works for debug ? On the other hand, we force the drain by Peeking above so :shrug:

Reply:
yeah, DCHECK here is more like a helpful comment